be/src/runtime/data_stream_recvr.cc

```diff
@@ -148,6 +148,8 @@ class DataStreamRecvr::SenderQueue {
 
     typedef std::list<ChunkItem> ChunkQueue;
     ChunkQueue _chunk_queue;
+    bool _is_pipeline_level_shuffle = false;
+    std::vector<bool> _has_chunks_per_driver_sequence;
```

Comment:
Maybe we can use `std::vector<std::atomic<bool>>`.

Reply:
I think strong consistency is not required here

Reply:
> I think strong consistency is not required here

@liuyehcf Could the non-atomic read increase the ineffective driver process loop?

Reply:
@liuyehcf Could this delay the time, when the `ExchangeSourceOperator` becomes ready from blocked?

Reply:
May delay, within 100 instruction cycles, and I think it is accetable. Atomic will reduce the  efficiency of cpu pipeline execution.